### PR TITLE
chore: workflow improvements

### PR DESCRIPTION
**Pull request by aljo242**
## 1. Summary

Add two new workflows that are both run once a week:
- long sim test (runs full `make-test-sim-non-determinism`)
- markdown link checker (lets us know if any links break over time)

Removes the `push` tag for the `buildtest` workflow.  This was  causing workflow duplicates to be run.

Adds build and test rules to run on `darwin/arm64` (apple silicon) and `darwin/amd64`

### CODE REVIEW

File: .github/workflows/simulation.yml [hunk 1]
```diff
@@ -1,17 +1,14 @@
 name: simulate quicksilver
 
 on:
-  pull_request:
-    branches:
-      - main
+  schedule:
+    - cron: '0 0 * * 0' # run once a week
```
Comment (ajansari95) [inline .github/workflows/simulation.yml:5]: can we do it based on some trigger? if we want to force run on some pe?

Comment (aljo242) [reply .github/workflows/simulation.yml:5]: So I moved the `run-sim` label trigger to `buildtest.yaml`.

This new scheduled job is an additional run of a really long simulation (more blocks and larger blocksize).

Comment (aljo242) [reply .github/workflows/simulation.yml:5]: Trigger still exists, just got moved to another file

**Review by joe-bowman — DISMISSED** [review]
lgtm

**Review by ajansari95 — APPROVED** [review]
LGTM

**Review by joe-bowman — APPROVED** [review]
lgtm

**Pull request merged**